### PR TITLE
Updated docs to include uint

### PR DESCRIPTION
**Pull request by jbednar**
- Documented the switch to uint for count aggregates
- Updated the way-obsolete list of aggregate types in 2_Points.ipynb; had only listed a few of the many now available.
- Replaced obsolete reference to `count_cat` with `by`

### CODE REVIEW

File: datashader/reductions.py [hunk 1]
```diff
@@ -216,7 +216,7 @@ def finalize(bases, cuda=False, **kwargs):
         return finalize
 
 class count(OptionalFieldReduction):
-    """Count elements in each bin.
+    """Count elements in each bin, returning the result as a unint32.
```
Comment (philippjfr) [inline datashader/reductions.py:219]:
```suggestion
    """Count elements in each bin, returning the result as a uint32.
```

File: examples/getting_started/2_Pipeline.ipynb [hunk 2]
```diff
@@ -145,34 +145,45 @@
...
     "  \n",
     "**`any(column=None)`**:\n",
     "  the bin is set to 1 if any datapoint maps to it, and 0 otherwise.\n",
     "  \n",
     "**`sum(column)`**:\n",
     "  add the value of the given column for this datapoint to a running total for this bin.\n",
     "   \n",
-    "**`count_cat(column)`**:\n",
-    "  given a bin with categorical data (i.e., [Pandas' `categorical` datatype](https://pandas-docs.github.io/pandas-docs-travis/categorical.html)), count each category separately, adding the given datapoint to an appropriate category within this bin.  These categories can later be collapsed into a single count if needed; see example below.\n",
+    "**`by(column)`**:\n",
```
Comment (philippjfr) [inline examples/getting_started/2_Pipeline.ipynb:182]: The signature for `by` is different, not sure how best to express `by(column, agg)`.

Comment (jbednar) [reply examples/getting_started/2_Pipeline.ipynb:182]: Good point; need to update that too.

Comment (jbednar) [inline examples/getting_started/2_Pipeline.ipynb:182]:
```suggestion
    "**`by(column, reduction)`**:\n",
```

**Pull request merged**